pkg/gcp/actuator/actuator.go

```diff
@@ -348,37 +431,59 @@ func (a *Actuator) needsUpdate(ctx context.Context, cr *minterv1.CredentialsRequ
 		// check if service account key in secret is still available
 		keyID, err := a.loadExistingSecretKeyAuthID(cr)
 		if err != nil {
-			return true, err
+			return serviceAPIsEnabled, true, err
 		}
 
 		keyExists, err := serviceAccountKeyExists(readClient, gcpStatus.ServiceAccountID, keyID, logger)
 		if err != nil {
 			logger.WithError(err).Error("error checking whether service account keys exists")
-			return true, err
+			return serviceAPIsEnabled, true, err
 		}
 		if !keyExists {
-			return true, nil
+			return serviceAPIsEnabled, true, nil
 		}
 
 		// TODO: add tagging check once we decide on tagging scheme
 
 		// check that the current permissions match what is being requested
 		needPermissionsUpdate, err := serviceAccountNeedsPermissionsUpdate(readClient, gcpStatus.ServiceAccountID, gcpSpec.PredefinedRoles)
 		if err != nil {
-			return true, fmt.Errorf("error determining whether policy binding update is needed: %v", err)
+			return serviceAPIsEnabled, true, fmt.Errorf("error determining whether policy binding update is needed: %v", err)
 		}
 		if needPermissionsUpdate {
 			logger.Debug("detected need for policy update")
-			return true, nil
+			return serviceAPIsEnabled, true, nil
 		}
 
 	} else {
 		// passthrough-specifc check here
-		return true, fmt.Errorf("PASSTHROUGH NOT IMPLEMENTED")
+
+		// build client with the target secret creds, then check permissions
+		testClient, err := a.buildClientFromCRSecret(cr)
```

Comment:
If we're doing passthrough here, why do we go to the target secret rather than using the root creds?

Just thinking about scenario where the root creds have been rotated and the old ones are no longer valid,  would be fail here because the target secret is still on the old creds and then we fail to copy over the new? (if so unit test would be good here)

Reply:
This is due to the dynamic permissions checking for GCP (as opposed to the static-mode we implemented for AWS). 'passthrough' basically reads in GCP as 'these creds have the potential to be passed through depending on the permissions being requested' versus in AWS where we are blindly assuming that the AWS secret with the passthrough annotation is good enough (based on the static list of perms in https://github.com/openshift/cloud-credential-operator/blob/master/pkg/aws/utils.go#L35 .

The thinking was that when we're checking whether an update is needed, if there is an existing secret, and it already has good-enough permissions to satisfy the credRequest, then just leave the secret alone as there is no update needed.

Reply:
To be clear, we could absolutely check against the cluster root creds, and then check whether the cluster root creds content is the same as the already-existing CR secret content, and decide needsUpdate() that way.

Reply:
If the existing target secret fails, then we'd return that we do indeed need to update, and that code path will handle checking the root creds before syncing the new secret contents.

Reply:
Re-read your original comment, and for cert rotation, you're absolutely right. We should check the root creds, and then make sure those are synced to the CR target secret. Will update.

Reply:
ok. changed the logic to check the root creds, and then whether the CR target secret data is out-of-date.